src/Symfony/Component/HttpClient/CachingHttpClient.php

```diff
@@ -106,41 +390,313 @@ public function request(string $method, string $url, array $options = []): Respo
+     */
+    private function getCurrentAge(array $headers): int
+    {
+        return (int) ($headers['age'][0] ?? 0);
+    }
+
+    /**
+     * Calculates the expiration time of the response given the maximum age.
+     *
+     * @param int|null $maxAge the maximum age of the response in seconds, or null if it cannot be determined
+     *
+     * @return int|null the expiration time of the response as a Unix timestamp, or null if the maximum age is null
+     */
+    private function calculateExpiresAt(?int $maxAge): ?int
+    {
+        if (null === $maxAge) {
+            return null;
+        }
+
+        return time() + $maxAge;
     }
 
-    public function reset(): void
+    /**
+     * Checks if the server response is cacheable according to the HTTP 1.1
+     * specification (RFC 9111).
+     *
+     * This function will return true if the server response can be cached,
+     * false otherwise.
+     *
+     * @param int                            $statusCode      the HTTP status code of the response
+     * @param array<string, string|string[]> $requestHeaders  the HTTP request headers
+     * @param array<string, string|string[]> $responseHeaders the HTTP response headers
+     * @param array<string, string|true>     $cacheControl    an array of parsed Cache-Control directives
+     *
+     * @return bool true if the response is cacheable, false otherwise
+     */
+    private function isServerResponseCacheable(int $statusCode, array $requestHeaders, array $responseHeaders, array $cacheControl): bool
     {
-        if ($this->client instanceof ResetInterface) {
-            $this->client->reset();
+        // no-store => skip caching
+        if (isset($cacheControl['no-store'])) {
+            return false;
         }
+
+        if (
+            $this->sharedCache
+            && !isset($cacheControl['public']) && !isset($cacheControl['s-maxage']) && !isset($cacheControl['must-revalidate'])
+            && isset($requestHeaders['authorization'])
+        ) {
+            return false;
+        }
+
+        if ($this->sharedCache && isset($cacheControl['private'])) {
+            return false;
+        }
+
+        // Conditionals require an explicit expiration
+        if (\in_array($statusCode, self::CONDITIONALLY_CACHEABLE_STATUS_CODES, true)) {
+            return $this->hasExplicitExpiration($responseHeaders, $cacheControl);
+        }
+
+        return \in_array($statusCode, self::CACHEABLE_STATUS_CODES, true);
+    }
+
+    /**
+     * Checks if the response has an explicit expiration.
+     *
+     * This function will return true if the response has an explicit expiration
+     * time specified in the headers or in the Cache-Control directives,
+     * false otherwise.
+     *
+     * @param array<string, string|string[]> $headers      the HTTP response headers
+     * @param array<string, string|true>     $cacheControl an array of parsed Cache-Control directives
```

Comment:
I suggest removing the description part of those tags. The argument names are already clear enough IMO.

Reply:
adressed in https://github.com/symfony/symfony/pull/59576/commits/b3154e1c1ecf2ad2b1c7b33e95549af3c79e3f4b

Reply:
adressed in https://github.com/symfony/symfony/pull/59576/commits/04d24d392fb6669061e7c01f9f9b270e5e15bdac